File: Practicas/Qytetet/QytetetRuby/lib/calle.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative "casilla"

module ModeloQytetet
  class Calle < Casilla

    attr_accessor :titulo
    attr_reader :tipo

    @tipo = TipoCasilla::CALLE
      
    def initialize(numero_c, t)
      super(numero_c, t.precio_compra)
      @titulo = t
    end

    def asignar_propietario(jugador)
      @titulo.propietario = jugador
      @titulo
    end

    def pagar_alquiler
      coste_alquiler = @titulo.pagar_alquiler
      coste_alquiler
    end

    def propietario_encarcelado
      @titulo.propietario_encarcelado
    end

    def soy_edificable
      true
    end

    def tengo_propietario
      @titulo.tengo_propietario
    end

    def to_s
      super.to_s + ", tipo=TipoCasilla::CALLE, titulo=#{@titulo}"
    end
  
  end
end

File: Practicas/Qytetet/QytetetJava/src/modeloqytetet/Calle.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package modeloqytetet;

/**
 *
 * @author enrique
 */
public class Calle extends Casilla {
    private TituloPropiedad titulo;
    public Calle(int numeroCasilla, TituloPropiedad titulo)
    {
        super(numeroCasilla, titulo.getPrecioCompra());
        this.titulo = titulo;
    }
    
    public TituloPropiedad asignarPropietario(Jugador jugador){
        titulo.setPropietario(jugador);
        return titulo;
    }
    
    public void setTitulo(TituloPropiedad titulo) {
        this.titulo = titulo;
    }
    
    protected TituloPropiedad getTitulo() {
        return titulo;
    }
    
    public int pagarAlquiler(){
        int costeAlquiler = titulo.pagarAlquiler();
        return costeAlquiler;
    }
    
    boolean propietarioEncarcelado(){
        return titulo.propietarioEncarcelado();
    }
    
    @Override
    protected boolean soyEdificable(){
        return true;
    }
    
    public boolean tengoPropietario(){
        return titulo.tengoPropietario();
    }
    
    @Override
    public TipoCasilla getTipo()
    {
        return TipoCasilla.CALLE;
    }
    
    @Override
    public String toString()
    {
        return super.toString() + ", tipo=" + getTipo() + ", titulo=" + titulo + "}\n";
    }
}
